### ISSUE TITLE
Do not reveal server details in error messages

### ISSUE THREAD

**Issue by lars-t-hansen** [NAICNO/Jobanalyzer#456]
With better error reporting (#449, #447) comes the risk of revealing detail it would be better not to reveal.  Obviously the server is behind firewall, has limited logins, and secrets are stored in protected directories, but defense in depth is a thing.  Consider a message like this, from `naicreport load` when provided with a bad data directory:
```
NAICREPORT FAILED
While running sonalyze
exit status 1
With stderr:
ERROR: Not a viable log directory: /home/larstha/sonar/data/nothere
```
This reveals the configuration of the server (and the identity of the login running the server).  With naicreport this is fairly innocent, since the error output goes to cron and the cron message goes to a specific admin's email, but 90% of the value during production would be had if the error would have been simply "Not a viable log directory" (or more properly, "data directory") - this allows users to report bugs and admins to know what to look for.

**Comment by lars-t-hansen** [comment]
I have audited our code as part of #447 and fixed a couple of obvious problems in the Rust code.  I don't think the Rust libraries reveal paths in error messages, but it's unknown if other libraries.

I have not tried to do anything about the Go code as part of that, but mostly that code currently runs only on the server and error messages won't be surfaced directly to users (not yet, see #451).

**Comment by lars-t-hansen** [comment]
Fixed by patch for #447.
